packages/next/build/swc/src/next_dynamic.rs

```diff
@@ -0,0 +1,174 @@
+  dynamic_binding: JsWord,
+}
+
+impl Fold for NextDynamicPatcher {
+  fn fold_import_decl(&mut self, decl: ImportDecl) -> ImportDecl {
+    let ImportDecl {
+      ref src,
+      ref specifiers,
+      ..
+    } = decl;
+    if &src.value == "next/dynamic" {
+      for specifier in specifiers {
+        if let ImportSpecifier::Default(default_specifier) = specifier {
+          self.dynamic_binding = default_specifier.local.sym.clone();
+        }
+      }
+    }
+
+    decl
+  }
+
+  fn fold_call_expr(&mut self, mut expr: CallExpr) -> CallExpr {
+    if let ExprOrSuper::Expr(i) = &expr.callee {
+      if let Expr::Ident(Ident { sym, .. }) = &**i {
+        if sym == &self.dynamic_binding {
+          let mut import_specifier = None;
+          if let Expr::Arrow(ArrowExpr {
+            body: BlockStmtOrExpr::Expr(e),
+            ..
+          }) = &*expr.args[0].expr
+          {
+            if let Expr::Call(CallExpr { args: a, .. }) = &**e {
+              if let Expr::Lit(Lit::Str(Str { value, .. })) = &*a[0].expr {
+                import_specifier = Some(value.clone());
+              }
+            }
+          }
+
+          let file_from = self.filename.clone().to_string() + " -> ";
+
+          // loadableGenerated: {
+          //   webpack: () => [require.resolveWeak('../components/hello')],
+          //   modules:
+          // ["/project/src/file-being-transformed.js -> " + '../components/hello'] }
+          let generated = Box::new(Expr::Object(ObjectLit {
+            span: DUMMY_SP,
+            props: vec![
+              PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+                key: PropName::Ident(Ident::new("webpack".into(), DUMMY_SP)),
+                value: Box::new(Expr::Arrow(ArrowExpr {
+                  params: vec![],
+                  body: BlockStmtOrExpr::Expr(Box::new(Expr::Array(ArrayLit {
+                    elems: vec![Some(ExprOrSpread {
+                      expr: Box::new(Expr::Call(CallExpr {
+                        callee: ExprOrSuper::Expr(Box::new(Expr::Member(MemberExpr {
+                          obj: ExprOrSuper::Expr(Box::new(Expr::Ident(Ident {
+                            sym: js_word!("require"),
+                            span: DUMMY_SP,
+                            optional: false,
+                          }))),
+                          prop: Box::new(Expr::Ident(Ident {
+                            sym: "resolveWeak".into(),
+                            span: DUMMY_SP,
+                            optional: false,
+                          })),
+                          computed: false,
+                          span: DUMMY_SP,
+                        }))),
+                        args: vec![ExprOrSpread {
+                          expr: Box::new(Expr::Lit(Lit::Str(Str {
+                            value: self.filename.clone().to_string().into(),
+                            span: DUMMY_SP,
+                            kind: StrKind::Normal {
```

Comment:
This should be `StrKind::Synthesized` or `Default::default()` because `value` of this `Str` didn't come from js source code.